p2p/src/authenticated/discovery/metrics.rs

```diff
@@ -18,6 +18,7 @@ impl Peer {
 
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
 pub enum MessageType {
+    Greeting,
```

Comment:
for consistency, let's have the order of these variants match the order of `Payload`